src/BuildScriptGeneratorCli/Commands/BuildCommand.cs

```diff
@@ -424,8 +424,21 @@ internal override IServiceProvider TryGetServiceProvider(IConsole console)
             SetPlatformVersion(PlatformName, PlatformVersion);
 
             commandLineConfigSource.Set(SettingsKeys.CreatePackage, ShouldPackage.ToString());
-            commandLineConfigSource.Set(SettingsKeys.RequiredOsPackages, OsRequirements);
-            commandLineConfigSource.Set(SettingsKeys.AppType, AppType);
+
+            if (!string.IsNullOrEmpty(OsRequirements))
+            {
+                commandLineConfigSource.Set(SettingsKeys.RequiredOsPackages, OsRequirements);
```

Comment:
nit: add a `SetIfNotEmpty` to avoid repeating all these ifs?

Reply:
Sure, will do